### PR TITLE
docs: v6 upgrading notes

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
This include some notes about the migration guide based on https://github.com/Unleash/unleash/issues/4380

Initially, we don't expect any breaking change provided users are not using deprecated functionality.

Preview notes: https://unleash-docs-git-v6-upgrade-notes-unleash-team.vercel.app/using-unleash/deploy/upgrading-unleash#upgrading-to-v6-from-v5

### CODE REVIEW

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 1]
```diff
@@ -5,6 +5,48 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
 
+## Upgrading to v6 from v5
+
+[Unleash v6](https://github.com/Unleash/unleash/issues/4380) was released on June 6th, 2024. **We expect this upgrade to be straightforward**, provided you followed the previous migration guides in this page. We're removing some features that were deprecated in version 5, new deprecated features that will continue to be supported during v6. Here's the list of the outstanding changes you need to take into account when migrating from v5 to v6:
+
+### Remove passport libs from the official Open Source Docker distribution
+
+*If you're not using the official open source image, you can safely ignore this change.* 
+
+The [official open source docker image](https://hub.docker.com/r/unleashorg/unleash-server) stop including these libraries which are required for SSO configuration. If you're using this feature, there is a [community image](https://github.com/Unleash/unleash-docker-community) which adds these dependencies.
```
Comment (ivarconr) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:16]:
```suggestion
The [official open source docker image](https://hub.docker.com/r/unleashorg/unleash-server) stop including these libraries which are custom authentication. If you're using this feature, there is a [community image](https://github.com/Unleash/unleash-docker-community) which adds these dependencies. 
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 2]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
 
+## Upgrading to v6 from v5
+
+[Unleash v6](https://github.com/Unleash/unleash/issues/4380) was released on June 6th, 2024. **We expect this upgrade to be straightforward**, provided you followed the previous migration guides in this page. We're removing some features that were deprecated in version 5, new deprecated features that will continue to be supported during v6. Here's the list of the outstanding changes you need to take into account when migrating from v5 to v6:
+
+### Remove passport libs from the official Open Source Docker distribution
+
+*If you're not using the official open source image, you can safely ignore this change.* 
+
+The [official open source docker image](https://hub.docker.com/r/unleashorg/unleash-server) stop including these libraries which are custom authentication. If you're using this feature, there is a [community image](https://github.com/Unleash/unleash-docker-community) which adds these dependencies.
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:16]:
```suggestion
The [official open source Docker image](https://hub.docker.com/r/unleashorg/unleash-server) will no longer include these libraries, which were used for custom authentication. If you're using this feature, there is a [community image](https://github.com/Unleash/unleash-docker-community) which includes these dependencies. 
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 3]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
 
 Generally, the intention is that `unleash-server` should always provide support for clients one major version lower than the current one. This should make it possible to upgrade `unleash` gradually.
 
+## Upgrading to v6 from v5
+
+[Unleash v6](https://github.com/Unleash/unleash/issues/4380) was released on June 6th, 2024. **We expect this upgrade to be straightforward**, provided you followed the previous migration guides in this page. We're removing some features that were deprecated in version 5, new deprecated features that will continue to be supported during v6. Here's the list of the outstanding changes you need to take into account when migrating from v5 to v6:
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:10]:
```suggestion
[Unleash v6](https://github.com/Unleash/unleash/issues/4380) was released on June 6th, 2024. **We expect this upgrade to be straightforward**, provided you have followed the previous migration guides on this page. We're removing some features that were deprecated in version 5, and deprecating some features that will continue to be supported during v6. Here's the list of the outstanding changes you need to take into account when migrating from v5 to v6:
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 2 above)
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:16]:
```suggestion
The [official open source docker image](https://hub.docker.com/r/unleashorg/unleash-server) no longer includes these custom authentication libraries. If you're using this feature, there is a [community image](https://github.com/Unleash/unleash-docker-community) that contains (and will continue to contain) these dependencies. 
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 4]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+### Drop support for postgres versions 10, 11 and 12
+
+*If you're using postgres 13 or above you can safely ignore this change*
+
+Unleash v6 will output an error message when starting with an unsuported version but will continue to run. We recommend upgrading to a supported version as soon as possible.
+
+### Update to Node.js version 20+
+
+Unleash v6 drops support Node.js versions below 20, which is the [current active LTS at the time of release](https://github.com/nodejs/Release/blob/6209d04302e62156b964a605f619283582334c95/README.md#release-schedule).
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:26]:
```suggestion
Unleash v6 drops support for Node.js versions below 20, which is the [active LTS at the time of release](https://github.com/nodejs/Release/blob/6209d04302e62156b964a605f619283582334c95/README.md#release-schedule).
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 5]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+### Remove /edge/metrics endpoint
+
+*If you're not using edge, you can safely ignore this change*
+
+If you're using edge to connect to Unleash, make sure to first upgrade unleash-edge to version 19.1.3 or above.
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:32]: What's the impact of not upgrading edge? You'll just not get metrics?

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 6]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+### Remove legacy "/api/feature
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:34]:
```suggestion
### Remove legacy `/api/feature` endpoint
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 7]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+If you're still using this endpoint, checkout the [deprecation notice in v4](#4-legacy-v2-routes-removed)
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:36]:
```suggestion
If you're still using this endpoint, check out the [deprecation notice in v4](#4-legacy-v2-routes-removed)
```

Comment (thomasheartman) [reply website/docs/using-unleash/deploy/upgrading-unleash.md:36]: Also, what do they need to do if they are using it? It's not really clear from that paragraph you link to 😅  Maybe we don't have a clear path, and that's okay too, but I think it'd be nice if it was clearer what to do.

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 8]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+### Remove deprecated [old import service](../../how-to/how-to-import-export)
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:38]: I'd remove "old" from the heading here and clarify in the body that this applies to the deprecated import/export functionality and not the new one. We should make it clear which one it is 

```suggestion
### Remove deprecated [import service](../../how-to/how-to-import-export)
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 9]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+Unleash v6 drops support for file based import at startup. If you're using this functionality, please check out the issue about adding this functionality to the current import service ([#7128](https://github.com/Unleash/unleash/issues/7128))
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:40]: Check out the issue and do what? Chime in? Monitor it for completion?

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 10]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+If you're using that endpoint for a custom integration, you should migrate to the [new import service](../../how-to/how-to-environment-import-export) (api docs are [here](../../reference/api/unleash/import-export)). For a comparisson between the two, please check the notes at the bottom of [this page](../../how-to/how-to-environment-import-export#environment-importexport-vs-the-instance-importexport-api)
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:42]: Just making sure the link text makes sense out of context

```suggestion
If you're using that endpoint for a custom integration, you should migrate to the [new import service](../../how-to/how-to-environment-import-export) ([new import/export api docs](../../reference/api/unleash/import-export)). 
For a comparison between the two, refer to the [environment import/export vs instance import/export](../../how-to/how-to-environment-import-export#environment-importexport-vs-the-instance-importexport-api) section at the bottom of this page.
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 11]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+### Custom strategies deprecation
+
+The downside of using a custom strategies is that you need to distribute the code of your strategy with Unleash client SDK while on the other hand, [strategy constraints](../../reference/strategy-constraints) work without any extra code or maintenance. In most cases [strategy constraints](../../reference/strategy-constraints) provides enough control without the additional complexity.
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:46]:
```suggestion
The downside of using a custom strategies is that you need to write and distribute the code for your strategy to your Unleash client SDKs. On the other hand, [strategy constraints](../../reference/strategy-constraints) work without any extra code or maintenance. In most cases [strategy constraints](../../reference/strategy-constraints) provides all the control you need without the additional complexity. 
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 12]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+If you're using that endpoint for a custom integration, you should migrate to the [new import service](../../how-to/how-to-environment-import-export) (api docs are [here](../../reference/api/unleash/import-export)). For a comparisson between the two, please check the notes at the bottom of [this page](../../how-to/how-to-environment-import-export#environment-importexport-vs-the-instance-importexport-api)
+
+### Custom strategies deprecation
+
+The downside of using a custom strategies is that you need to distribute the code of your strategy with Unleash client SDK while on the other hand, [strategy constraints](../../reference/strategy-constraints) work without any extra code or maintenance. In most cases [strategy constraints](../../reference/strategy-constraints) provides enough control without the additional complexity. 
+
+If you can't accomplish the same functionality with strategy constraints, please, let us know about your use case on [Slack](https://slack.unleash.run/)
```
Comment (thomasheartman) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:48]:
```suggestion
If you can't accomplish the same functionality with strategy constraints, please let us know about your use case on [Slack](https://slack.unleash.run/)
```

**Review by thomasheartman — APPROVED** [review]
Nice work! Got a few small suggestions here and there, but it's all pretty minor stuff 👏🏼 👏🏼 👏🏼

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 3 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:10]:
```suggestion
[Unleash v6](https://github.com/Unleash/unleash/issues/4380) was released on June 6th, 2024. **We expect this upgrade to be straightforward**, provided you followed the previous migration guides in this page. We're removing some features that were deprecated in version 5 and marking some new features as deprecated, but these will continue to be supported during version 6. Here's the list of the outstanding changes you need to take into account when migrating from v5 to v6:
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 13]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
 
 Generally, the intention is that `unleash-server` should always provide support for clients one major version lower than the current one. This should make it possible to upgrade `unleash` gradually.
 
+## Upgrading to v6 from v5
+
+[Unleash v6](https://github.com/Unleash/unleash/issues/4380) was released on June 6th, 2024. **We expect this upgrade to be straightforward**, provided you followed the previous migration guides in this page. We're removing some features that were deprecated in version 5, new deprecated features that will continue to be supported during v6. Here's the list of the outstanding changes you need to take into account when migrating from v5 to v6:
+
+### Remove passport libs from the official Open Source Docker distribution
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:12]:
```suggestion
### Remove Passport libs from the official open source Docker distribution
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 14]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+*If you're not using the official open source image, you can safely ignore this change.* 
+
+The [official open source docker image](https://hub.docker.com/r/unleashorg/unleash-server) stop including these libraries which are custom authentication. If you're using this feature, there is a [community image](https://github.com/Unleash/unleash-docker-community) which adds these dependencies. 
+
+### Drop support for postgres versions 10, 11 and 12
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:18]:
```suggestion
### Drop support for PostgreSQL versions 10, 11 and 12
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 15]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+*If you're using postgres 13 or above you can safely ignore this change*
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:20]:
```suggestion
*If you're using PostgreSQL 13 or above you can safely ignore this change.*
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 16]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+Unleash v6 will output an error message when starting with an unsuported version but will continue to run. We recommend upgrading to a supported version as soon as possible.
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:22]:
```suggestion
Unleash v6 will output an error message when starting with an unsupported version but continue to run. We recommend upgrading to a supported version as soon as possible.
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 4 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:26]:
```suggestion
Unleash v6 drops support for Node.js versions below 20, which is the [current active LTS at the time of release](https://github.com/nodejs/Release/blob/6209d04302e62156b964a605f619283582334c95/README.md#release-schedule).
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 17]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+### Update to Node.js version 20+
+
+Unleash v6 drops support Node.js versions below 20, which is the [current active LTS at the time of release](https://github.com/nodejs/Release/blob/6209d04302e62156b964a605f619283582334c95/README.md#release-schedule).
+
+### Remove /edge/metrics endpoint
+
+*If you're not using edge, you can safely ignore this change*
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:30]:
```suggestion
*If you're not using Unleash Edge, you can safely ignore this change.*
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 5 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:32]:
```suggestion
If you're using Unleash Edge to connect to Unleash, make sure to first upgrade Unleash Edge to version 19.1.3 or above.
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 6 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:34]:
```suggestion
### Remove legacy /api/feature endpoint
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 7 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:36]:
```suggestion
If you're still using this endpoint, checkout the [deprecation notice in v4](#4-legacy-v2-routes-removed).
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 9 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:40]:
```suggestion
Unleash v6 drops support for file based import at startup. If you're using this functionality, please check out the issue about adding this functionality to the current import service ([#7128](https://github.com/Unleash/unleash/issues/7128)).
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 10 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:42]:
```suggestion
If you're using that endpoint for a custom integration, you should migrate to the [new import service](../../how-to/how-to-environment-import-export) (api docs are [here](../../reference/api/unleash/import-export)). For a comparison between the two, please check the notes at the bottom of [this page](../../how-to/how-to-environment-import-export#environment-importexport-vs-the-instance-importexport-api).
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 11 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:46]:
```suggestion
Using custom strategies has the drawback of requiring you to distribute the strategy's code with the Unleash client SDK. In contrast, [strategy constraints](../../reference/strategy-constraints) function without needing additional code or maintenance. In most cases, [strategy constraints](../../reference/strategy-constraints) offer sufficient control without added complexity.
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 12 above)
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:48]:
```suggestion
If you can't accomplish the same functionality with strategy constraints, please let us know about your use case on [Slack](https://slack.unleash.run/).
```

File: website/docs/using-unleash/deploy/upgrading-unleash.md [hunk 18]
```diff
@@ -5,6 +5,52 @@ import VideoContent from '@site/src/components/VideoContent.jsx'
...
+
+If you're still using this endpoint, checkout the [deprecation notice in v4](#4-legacy-v2-routes-removed)
+
+### Remove deprecated [old import service](../../how-to/how-to-import-export)
+
+Unleash v6 drops support for file based import at startup. If you're using this functionality, please check out the issue about adding this functionality to the current import service ([#7128](https://github.com/Unleash/unleash/issues/7128))
+
+If you're using that endpoint for a custom integration, you should migrate to the [new import service](../../how-to/how-to-environment-import-export) (api docs are [here](../../reference/api/unleash/import-export)). For a comparisson between the two, please check the notes at the bottom of [this page](../../how-to/how-to-environment-import-export#environment-importexport-vs-the-instance-importexport-api)
+
+### Custom strategies deprecation
```
Comment (nunogois) [inline website/docs/using-unleash/deploy/upgrading-unleash.md:44]: More aligned with the other headers? Either this or change the other ones.

```suggestion
### Deprecate custom strategies
```

**Review by nunogois — APPROVED** [review]
👍

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 5 above)
Comment (gastonfournier) [reply website/docs/using-unleash/deploy/upgrading-unleash.md:32]: Indeed

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 2 above)
Comment (gastonfournier) [reply website/docs/using-unleash/deploy/upgrading-unleash.md:16]: Adding this to a commit of my own

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 5 above)
Comment (gastonfournier) [reply website/docs/using-unleash/deploy/upgrading-unleash.md]: Adding this to my commit

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 3 above)
Comment (gastonfournier) [reply website/docs/using-unleash/deploy/upgrading-unleash.md]: Adding it to my next commit

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 7 above)
Comment (gastonfournier) [reply website/docs/using-unleash/deploy/upgrading-unleash.md]: Added to my commit

File: website/docs/using-unleash/deploy/upgrading-unleash.md (hunk 11 above)
Comment (gastonfournier) [reply website/docs/using-unleash/deploy/upgrading-unleash.md]: Added to my commit

**Review by ivarconr — APPROVED** [review]

**Pull request merged**